# OS vs. Theme Preference (Test of Independence)

### Objective:
Analyze UI telemetry data from 350 desktop application users to determine if a user's Operating System dictates their Theme Preference (Dark Mode vs. Light Mode), or if the two categories are mathematically independent. 

**The Telemetry Matrix (Observed Data):**
*   **Windows:** 140 Dark Mode | 60 Light Mode
*   **Mac:** 32 Dark Mode | 68 Light Mode
*   **Linux:** 40 Dark Mode | 10 Light Mode

>Test Used: Chi-Square Test of Independence

>Performed by: Raghav Katyal on 30 September 2026

In [9]:
import numpy as np
from scipy import stats

In [10]:
# The columns are perfectly aligned: [Dark Mode, Light Mode].
user_base = np.array([
    [140,60],
    [32,68],
    [40,10]
])

In [11]:
#Running the test
chi_stat,p_value,_,_ = stats.chi2_contingency(user_base)

In [12]:
print(f"Chi-Square Score : {chi_stat:.2f}")
print(f"P-value : {p_value:.5f}")

Chi-Square Score : 49.53
P-value : 0.00000


>Observation : A Chi-Square score of 49.53 and p-value score of 0.00 shows strong evidence against the independence of data ; data fails the independence test. Choice of theme is in this example observed as highly dependent on OS type.